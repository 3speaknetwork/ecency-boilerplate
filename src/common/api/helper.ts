import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
