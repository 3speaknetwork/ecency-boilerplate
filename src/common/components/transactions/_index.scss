.transaction-list {
  .transaction-list-header {
    h2 {
      font-size: 18px;
      margin-bottom: 16px;
    }
  }

  .transaction-list-item {
    border-radius: 8px;
    box-sizing: content-box;
    display: flex;
    padding: 10px;
    font-size: 13px;

    @media (min-width: $lg-break) {
      font-size: 16px;
      padding: 16px;
    }

    .transaction-icon {
      align-items: center;
      border-radius: 50%;
      display: flex;
      height: 28px;
      justify-content: center;
      margin-right: 10px;
      width: 28px;

      svg {
        width: 16px;
        height: 16px;
      }

      @media (min-width: $lg-break) {
        font-size: 16px;
        height: 32px;
        width: 32px;
      }

      @include themify(day) {
        background: $white;
        color: $steel-grey;
      }

      @include themify(night) {
        background: $metallic-blue;
        color: lighten($metallic-blue, 40);
      }
    }

    .transaction-title {
      display: flex;
      flex-direction: column;
      font-weight: 500;
      width: 30%;

      .transaction-name {
        @include themify(day) {
          color: $steel-grey;
        }

        @include themify(night) {
          color: $metallic-blue;
        }
      }

      .transaction-date {
        font-size: 12px;

        @include themify(day) {
          color: $silver;
        }

        @include themify(night) {
          color: $warm-grey;
        }
      }
    }

    .transaction-numbers {
      color: $dark-sky-blue;
      flex-grow: 1;
      font-weight: 500;
      text-align: right;

      .number {
        margin-right: 4px;

        &:not(:last-child) {
          &:after {
            content: ",";
          }
        }
      }

      @media (min-width: $md-break) {
        margin-right: 10px;
      }

      @media (min-width: $lg-break) {
        margin-right: 30px;
      }
    }

    .transaction-details {
      color: $steel-grey;
      width: 30%;
      word-break: break-all;
      display: none;

      @media (min-width: $md-break) {
        display: block;
      }
    }

    &:nth-child(odd) {
      @include themify(day) {
        background: $white-three;
      }

      @include themify(night) {
        background: lighten($dark-six, 4);
      }
    }

    @include themify(day) {
      &:nth-child(even) {
        .transaction-icon {
          background-color: $white-three;
        }
      }
    }
  }
}
