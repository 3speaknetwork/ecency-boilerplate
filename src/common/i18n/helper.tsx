import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
