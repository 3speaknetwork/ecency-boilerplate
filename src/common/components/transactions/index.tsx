import React, { Component } from "react";

import moment from "moment";

import { DynamicProps } from "../../store/dynamic-props/types";
import { Transaction, Transactions } from "../../store/transactions/types";

import LinearProgress from "../linear-progress";

import parseAsset from "../../helper/parse-asset";
import parseDate from "../../helper/parse-date";
import { vestsToSp } from "../../helper/vesting";

import formattedNumber from "../../util/formatted-number";

import { ticketSvg, commentSvg, compareHorizontalSvg, cashSvg, reOrderHorizontalSvg } from "../../img/svg";

import { _t } from "../../i18n";

interface RowProps {
  dynamicProps: DynamicProps;
  transaction: Transaction;
}

export class TransactionRow extends Component<RowProps> {
  render() {
    const { dynamicProps, transaction: tr } = this.props;
    const { hivePerMVests } = dynamicProps;

    let flag = false;
    let icon = ticketSvg;
    let numbers = null;
    let details = null;

    if (tr.type === "curation_reward") {
      flag = true;

      numbers = <>{formattedNumber(vestsToSp(parseAsset(tr.reward).amount, hivePerMVests), { suffix: "HP" })}</>;
      details = (
        <span>
          <a href={`/curation/@${tr.comment_author}/${tr.comment_permlink}`}>@{tr.comment_author}/{tr.comment_permlink}</a>  
        </span>
      );
    }

    if (tr.type === "author_reward" || tr.type === "comment_benefactor_reward") {
      flag = true;

      const hbd_payout = parseAsset(tr.sbd_payout);
      const hive_payout = parseAsset(tr.steem_payout);
      const vesting_payout = parseAsset(tr.vesting_payout);
      numbers = (
        <>
          {hbd_payout.amount > 0 && (
            <span className="number">{formattedNumber(hbd_payout.amount, { suffix: "HBD" })}</span>
          )}
          {hive_payout.amount > 0 && (
            <span className="number">{formattedNumber(hive_payout.amount, { suffix: "HIVE" })}</span>
          )}
          {vesting_payout.amount > 0 && (
            <span className="number">
              {formattedNumber(vestsToSp(vesting_payout.amount, hivePerMVests), { suffix: "HP" })}{" "}
            </span>
          )}
        </>
      );

      details = (
        <span>
          <a href={`/reward/@${tr.author}/${tr.permlink}`}>@{tr.author}/{tr.permlink}</a>  
        </span>
      );
    }

    if (tr.type === "comment_benefactor_reward") {
      icon = commentSvg;
    }

    if (tr.type === "claim_reward_balance") {
      flag = true;

      const reward_hbd = parseAsset(tr.reward_sbd || tr.reward_hbd);
      const reward_hive = parseAsset(tr.reward_steem || tr.reward_hive);
      const reward_vests = parseAsset(tr.reward_vests);

      numbers = (
        <>
          {reward_hbd.amount > 0 && (
            <span className="number">{formattedNumber(reward_hbd.amount, { suffix: "HBD" })}</span>
          )}
          {reward_hive.amount > 0 && (
            <span className="number">{formattedNumber(reward_hive.amount, { suffix: "HIVE" })}</span>
          )}
          {reward_vests.amount > 0 && (
            <span className="number">
              {formattedNumber(vestsToSp(reward_vests.amount, hivePerMVests), { suffix: "HP" })}
            </span>
          )}
        </>
      );
    }

    if (tr.type === "transfer" || tr.type === "transfer_to_vesting") {
      flag = true;
      icon = compareHorizontalSvg;

      details = (
        <span>
          {tr.memo ? (
            <>
              {tr.memo} <br /> <br />
            </>
          ) : null}
          <strong>@{tr.from}</strong> -&gt; <strong>@{tr.to}</strong>
        </span>
      );

      numbers = <span className="number">{tr.amount}</span>;
    }

    if (tr.type === "withdraw_vesting") {
      flag = true;
      icon = cashSvg;

      const vesting_shares = parseAsset(tr.vesting_shares);
      numbers = (
        <span className="number">
          {formattedNumber(vestsToSp(vesting_shares.amount, hivePerMVests), { suffix: "HP" })}
        </span>
      );

      details = tr.acc ? (
        <span>
          <strong>@{tr.acc}</strong>
        </span>
      ) : null;
    }

    if (tr.type === "fill_order") {
      flag = true;
      icon = reOrderHorizontalSvg;

      numbers = (
        <span className="number">
          {tr.current_pays} = {tr.open_pays}
        </span>
      );
    }

    if (flag) {
      const transDate = parseDate(tr.timestamp);

      return (
        <div className="transaction-list-item">
          <div className="transaction-icon">{icon}</div>
          <div className="transaction-title">
            <div className="transaction-name">{_t(`transactions.type-${tr.type}`)}</div>
            <div className="transaction-date">{moment(transDate).fromNow()}</div>
          </div>
          <div className="transaction-numbers">{numbers}</div>
          <div className="transaction-details">{details}</div>
        </div>
      );
    }
    return null;
  }
}

interface Props {
  dynamicProps: DynamicProps;
  transactions: Transactions;
}

export default class TransactionList extends Component<Props> {
  render() {
    const { transactions } = this.props;
    const { list, loading } = transactions;

    // Top 50 transaction sorted by id
    const trList = list.slice(Math.max(list.length - 50, 0)).sort((a: any, b: any) => b.num - a.num);

    return (
      <div className="transaction-list">
        <div className="transaction-list-header">
          <h2>{_t("transactions.title")} </h2>
        </div>
        {loading && <LinearProgress />}
        {trList.map((x, k) => (
          <TransactionRow {...this.props} key={k} transaction={x} />
        ))}
      </div>
    );
  }
}
