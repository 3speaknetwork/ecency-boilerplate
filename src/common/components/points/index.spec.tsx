import React from "react";

import {UserPoints} from "./index";

import TestRenderer from "react-test-renderer";

import {createBrowserHistory} from "history";

import {initialState as transactionsInitialState} from "../../store/transactions/index";

import {globalInstance, pointTransactionsInstance} from "../../helper/test-helper";

jest.mock("moment", () => () => ({
    fromNow: () => "5 days ago",
}));

it("(1) Default Render", () => {
    const props = {
        history: createBrowserHistory(),
        global: globalInstance,
        activeUser: null,
        account: {
            name: "user1",
        },
        points: {
            points: "12.010",
            uPoints: "0.000",
            transactions: [...pointTransactionsInstance]
        },
        transactions: transactionsInitialState,
        addAccount: () => {
        },
        updateActiveUser: () => {

        }
    };

    const renderer = TestRenderer.create(<UserPoints {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
});

it("(2) With active user", () => {
    const props = {
        history: createBrowserHistory(),
        global: globalInstance,
        activeUser: {
            username: "user1",
            data: {
                name: "user1",
            }
        },
        account: {
            name: "user1",
        },
        points: {
            points: "12.010",
            uPoints: "0.000",
            transactions: [...pointTransactionsInstance]
        },
        transactions: transactionsInitialState,
        addAccount: () => {
        },
        updateActiveUser: () => {

        }
    };

    const renderer = TestRenderer.create(<UserPoints {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(3) Active user with unclaimed points", () => {
    const props = {
        history: createBrowserHistory(),
        global: globalInstance,
        activeUser: {
            username: "user1",
            data: {
                name: "user1",
            }
        },
        account: {
            name: "user1",
        },
        points: {
            points: "12.010",
            uPoints: "6.200",
            transactions: [...pointTransactionsInstance]
        },
        transactions: transactionsInitialState,
        addAccount: () => {
        },
        updateActiveUser: () => {

        }
    };

    const renderer = TestRenderer.create(<UserPoints {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
});
