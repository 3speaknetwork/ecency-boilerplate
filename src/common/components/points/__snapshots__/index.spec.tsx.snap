// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default Render 1`] = `
<div
  className="points-section"
>
  <div
    className="points"
  >
    <div
      className="point-name"
    >
      Ecency Points
    </div>
    <div
      className="points-val"
    >
      <div
        className="val"
      >
        12.010
         POINTS
      </div>
    </div>
    <div
      className="clearfix"
    />
  </div>
  <div
    className="point-reward-types"
  >
    <div
      className="point-reward-type"
      title="You can earn point by posting regularly. Posting gives you 15 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        15
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Each comment you make helps you to grow your audience and make friendship but also earns you 5 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        5
      </span>
    </div>
    <div
      className="point-reward-type"
      title="By voting you give reward to other creators and show your appreciation but also earn 0.01 x vote weight points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.3
      </span>
    </div>
    <div
      className="point-reward-type"
      title="  Share what post you like with your friends and earn 1 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Checking in on Ecency gives you 0.25 points and helps you stay connected with your friends."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.25
      </span>
    </div>
    <div
      className="point-reward-type"
      title="When you login into Ecency you are entitled to earn 100 points automatically."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,6A2,2 0 0,0 10,8A2,2 0 0,0 12,10A2,2 0 0,0 14,8A2,2 0 0,0 12,6M12,13C14.67,13 20,14.33 20,17V20H4V17C4,14.33 9.33,13 12,13M12,14.9C9.03,14.9 5.9,16.36 5.9,17V18.1H18.1V17C18.1,16.36 14.97,14.9 12,14.9Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        99+
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Consistent use of app gives you extra chances to earn more 10 points, be more active and earn more."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M0.41,13.41L6,19L7.41,17.58L1.83,12M22.24,5.58L11.66,16.17L7.5,12L6.07,13.41L11.66,19L23.66,7M18,7L16.59,5.58L10.24,11.93L11.66,13.34L18,7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        10
      </span>
    </div>
    <div
      className="point-reward-type"
      title="You can earn 1 point per day for each 100SP delegation."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
  </div>
  <div
    className="p-transaction-list"
  >
    <div
      className="transaction-list-header"
    >
      <h2>
        History
      </h2>
    </div>
    <div
      className="transaction-list-body"
    >
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for heartbeat
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.250
        </div>
      </div>
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for vote
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.750
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`(2) With active user 1`] = `
<div
  className="points-section"
>
  <div
    className="points"
  >
    <div
      className="point-name"
    >
      Ecency Points
    </div>
    <div
      className="points-val"
    >
      <div
        className="val"
      >
        12.010
         POINTS
      </div>
      <div
        className="custom-dropdown float-right"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="dropdown-btn "
        >
          <div
            className="label"
          >
            
          </div>
          <div
            className="menu-down"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7,10L12,15L17,10H7Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="clearfix"
    />
    <div
      className="unclaimed can-claim"
    >
      <div
        className="val"
      >
        <div
          className="val"
        >
          GET POINTS
        </div>
      </div>
      <a
        className="claim-btn "
        href="#"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M17,13H13V17H11V13H7V11H11V7H13V11H17M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="point-reward-types"
  >
    <div
      className="point-reward-type"
      title="You can earn point by posting regularly. Posting gives you 15 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        15
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Each comment you make helps you to grow your audience and make friendship but also earns you 5 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        5
      </span>
    </div>
    <div
      className="point-reward-type"
      title="By voting you give reward to other creators and show your appreciation but also earn 0.01 x vote weight points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.3
      </span>
    </div>
    <div
      className="point-reward-type"
      title="  Share what post you like with your friends and earn 1 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Checking in on Ecency gives you 0.25 points and helps you stay connected with your friends."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.25
      </span>
    </div>
    <div
      className="point-reward-type"
      title="When you login into Ecency you are entitled to earn 100 points automatically."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,6A2,2 0 0,0 10,8A2,2 0 0,0 12,10A2,2 0 0,0 14,8A2,2 0 0,0 12,6M12,13C14.67,13 20,14.33 20,17V20H4V17C4,14.33 9.33,13 12,13M12,14.9C9.03,14.9 5.9,16.36 5.9,17V18.1H18.1V17C18.1,16.36 14.97,14.9 12,14.9Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        99+
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Consistent use of app gives you extra chances to earn more 10 points, be more active and earn more."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M0.41,13.41L6,19L7.41,17.58L1.83,12M22.24,5.58L11.66,16.17L7.5,12L6.07,13.41L11.66,19L23.66,7M18,7L16.59,5.58L10.24,11.93L11.66,13.34L18,7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        10
      </span>
    </div>
    <div
      className="point-reward-type"
      title="You can earn 1 point per day for each 100SP delegation."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
  </div>
  <div
    className="p-transaction-list"
  >
    <div
      className="transaction-list-header"
    >
      <h2>
        History
      </h2>
    </div>
    <div
      className="transaction-list-body"
    >
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for heartbeat
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.250
        </div>
      </div>
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for vote
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.750
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`(3) Active user with unclaimed points 1`] = `
<div
  className="points-section"
>
  <div
    className="points"
  >
    <div
      className="point-name"
    >
      Ecency Points
    </div>
    <div
      className="points-val"
    >
      <div
        className="val"
      >
        12.010
         POINTS
      </div>
      <div
        className="custom-dropdown float-right"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="dropdown-btn "
        >
          <div
            className="label"
          >
            
          </div>
          <div
            className="menu-down"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7,10L12,15L17,10H7Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="clearfix"
    />
    <div
      className="unclaimed can-claim"
    >
      <div
        className="val"
      >
        <div
          className="val"
        >
          6.200
        </div>
      </div>
      <a
        className="claim-btn "
        href="#"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M17,13H13V17H11V13H7V11H11V7H13V11H17M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      className="clearfix"
    />
    <div
      className="get-estm"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        GET POINTS
      </a>
    </div>
  </div>
  <div
    className="point-reward-types"
  >
    <div
      className="point-reward-type"
      title="You can earn point by posting regularly. Posting gives you 15 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        15
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Each comment you make helps you to grow your audience and make friendship but also earns you 5 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        5
      </span>
    </div>
    <div
      className="point-reward-type"
      title="By voting you give reward to other creators and show your appreciation but also earn 0.01 x vote weight points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.3
      </span>
    </div>
    <div
      className="point-reward-type"
      title="  Share what post you like with your friends and earn 1 points."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Checking in on Ecency gives you 0.25 points and helps you stay connected with your friends."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        0.25
      </span>
    </div>
    <div
      className="point-reward-type"
      title="When you login into Ecency you are entitled to earn 100 points automatically."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,6A2,2 0 0,0 10,8A2,2 0 0,0 12,10A2,2 0 0,0 14,8A2,2 0 0,0 12,6M12,13C14.67,13 20,14.33 20,17V20H4V17C4,14.33 9.33,13 12,13M12,14.9C9.03,14.9 5.9,16.36 5.9,17V18.1H18.1V17C18.1,16.36 14.97,14.9 12,14.9Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        99+
      </span>
    </div>
    <div
      className="point-reward-type"
      title="Consistent use of app gives you extra chances to earn more 10 points, be more active and earn more."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M0.41,13.41L6,19L7.41,17.58L1.83,12M22.24,5.58L11.66,16.17L7.5,12L6.07,13.41L11.66,19L23.66,7M18,7L16.59,5.58L10.24,11.93L11.66,13.34L18,7Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        10
      </span>
    </div>
    <div
      className="point-reward-type"
      title="You can earn 1 point per day for each 100SP delegation."
    >
      <svg
        viewBox="0 0 24 24"
      >
        <path
          d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
          fill="currentColor"
        />
      </svg>
      <span
        className="reward-num"
      >
        1
      </span>
    </div>
  </div>
  <div
    className="p-transaction-list"
  >
    <div
      className="transaction-list-header"
    >
      <h2>
        History
      </h2>
    </div>
    <div
      className="transaction-list-body"
    >
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for heartbeat
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.250
        </div>
      </div>
      <div
        className="transaction-list-item"
      >
        <div
          className="transaction-icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          className="transaction-title"
        >
          <div
            className="transaction-name"
          >
            Points for vote
          </div>
          <div
            className="transaction-date"
          >
            5 days ago
          </div>
        </div>
        <div
          className="transaction-numbers"
        >
          0.750
        </div>
      </div>
    </div>
  </div>
</div>
`;
